### PR TITLE
Fixed web3-eth-abi typing

### PR DESCRIPTION
**Pull request by marcj**
The actual [code exports](https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-eth-abi/src/index.js#L442) as default a AbiCoder instance. In the previous version the class itself was exported via its name, which breaks runtime code.

Correctly used it looks like that:

```typescript
import abiCoder from 'web3-eth-abi';

const data = abiCoder.encodeParameters(...);
```

The new typings reflects that now correctly.

## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../CONTRIBUTIONS.md).

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Comment by GregTheGreek** [comment]
Thanks for the PR, I haven't tested this specifically yet, but from my understanding this doesn't quite solve it and we have a deeper issue that requires breaking the api.

AFAIIK Typescript doesn't have a way to export a instantiated classes. Specifically, abicoder looks like:
```
const coder = new AbiCoder();
module.exports = coder;
```

**Comment by marcj** [comment]
For me this solved it. We can not use the class AbiCoder in user land as you indicated since it isn't exported in the javascript file.

> AFAIIK Typescript doesn't have a way to export a instantiated classes

My PR provides precisely that.

> but from my understanding this doesn't quite solve it and we have a deeper issue that requires breaking the api

I don't see a deeper issue here. The published TS api is broken and unusable, so fixing it is the only way. No need to change the JS.

**Comment by GregTheGreek** [comment]
> For me this solved it. We can not use the class AbiCoder in user land as you indicated since it isn't exported in the javascript file.
> 
> > AFAIIK Typescript doesn't have a way to export a instantiated classes
> 
> My PR provides precisely that.
> 
> > but from my understanding this doesn't quite solve it and we have a deeper issue that requires breaking the api
> 
> I don't see a deeper issue here. The published TS api is broken and unusable, so fixing it is the only way. No need to change the JS.

Let me run this tomorrow, last time someone brought this up, it actually didn't solve the problem. But perhaps we were all mistaken!

**Comment by marcj** [comment]
👋 Alright, let me know if you need anything else.

**Comment by onchainguy-btc** [comment]
Would be awesome to see this merged, having the same issue 👐

**Comment by chancehudson** [comment]
I'd like to see this merged as well.

```
import AbiCoder from 'web3-eth-abi'
const encodeFunctionSignature = (AbiCoder as any).encodeFunctionSignature.bind(AbiCoder)
const decodeParameters = (AbiCoder as any).decodeParameters.bind(AbiCoder)
```

Ideally the export would simply be the functions, not a class. These are stateless utilities, why am I having to bind them?

I'll definitely take this PR as a fix though, it will let me remove the `any` cast.

**Comment by sambacha** [comment]
this is deff. not stale issue

**Comment by sambacha** [comment]
> This PR has been automatically marked as stale beacause it has not had recent activity. It will be closed in 7 days if no further activity occurs. Thank you for your contributions. If you believe this was a mistake, please comment.

lgtm

**Comment by marcj** [comment]
can you please merge that?

**Comment by Lucid36** [comment]
Fixed

**Comment by marcj** [comment]
Can you please merge that?

File: packages/web3-eth-abi/types/index.d.ts [hunk 1]
```diff
@@ -40,3 +40,6 @@ export class AbiCoder {
         topics: string[]
     ): { [key: string]: string };
 }
+
+declare var coder: AbiCoder;
```
Comment (avkos) [inline packages/web3-eth-abi/types/index.d.ts:44]:
```suggestion
declare const coder: AbiCoder;
```
I think you need to use `const` here

**Review by avkos — CHANGES_REQUESTED** [review]

**Review by nazarhussain — APPROVED** [review]
Apart from feedback from @avkos LGTM

File: packages/web3-eth-abi/types/index.d.ts (hunk 1 above)
Comment (marcj) [reply packages/web3-eth-abi/types/index.d.ts:44]: Why need? If it doesn't match you coding style, you can change it afterwards, or close and commit yourself. For me its only important that this bug gets fixed, not how.

Comment (avkos) [reply packages/web3-eth-abi/types/index.d.ts:44]: just fix it, please. and also check tests. They do not pass after your changes.

**Pull request closed without merging**